security/src/main/java/io/airlift/security/client/ClientSecurityConfig.java

```diff
@@ -0,0 +1,92 @@
+package io.airlift.security.client;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+
+public class ClientSecurityConfig
+{
+    private String krb5ccPath;
+    private String krb5Keytab;
+    private String krb5ConfPath;
+    private String krb5ClientPrincipal;
+    private boolean authenticationEnabled;
+    private String serviceName;
+
+    public String getKrb5ccPath()
+    {
+        return krb5ccPath;
+    }
+
+    @Config("http.client.authentication.negotiate.krb5cc-path")
+    @ConfigDescription("Set kerberos credential cache path")
+    public ClientSecurityConfig setKrb5ccPath(String krb5ccPath)
+    {
+        this.krb5ccPath = krb5ccPath;
+        return this;
+    }
+
+    public String getKrb5Keytab()
+    {
+        return krb5Keytab;
+    }
+
+    @Config("http.client.authentication.negotiate.krb5-keytab")
+    @ConfigDescription("Set kerberos key table path")
+    public ClientSecurityConfig setKrb5Keytab(String krb5Keytab)
+    {
+        this.krb5Keytab = krb5Keytab;
+        return this;
+    }
+
+    public String getKrb5ClientPrincipal()
+    {
+        return krb5ClientPrincipal;
+    }
+
+    @Config("http.client.authentication.negotiate.krb5-client-principal")
```

Comment:
Can you remind me what's the relationship between the client principal, the "client service name" (below) and the "server service name" (e.g, for systems that act both as servers and clients)?